processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -51,17 +54,87 @@ protected ListeningExecutorService delegate()
   }
 
   @Override
-  public <T> ListenableFuture<T> submit(Callable<T> tCallable)
+  public <T> ListenableFuture<T> submit(final Callable<T> tCallable)
   {
+    final Callable<T> wrappedPrioritizedCallable = new PrioritizedCallable<T>()
+    {
+      @Override
+      public int getPriority()
+      {
+        if (tCallable instanceof PrioritizedCallable) {
+          return ((PrioritizedCallable) tCallable).getPriority();
+        } else {
+          return 0;
+        }
+      }
+
+      @Override
+      public T call() throws Exception
+      {
+        final Long threadUserTime;
+        if (threadMXBean.isThreadCpuTimeEnabled()) {
+          threadUserTime = threadMXBean.getCurrentThreadUserTime();
+        } else {
+          threadUserTime = null;
+        }
+        final T retval = tCallable.call();
+        if (threadUserTime != null) {
+          emitter.emit(
+              metricBuilder
+                  .setDimension("priority", String.format("%d", getPriority()))
+                  .build("exec/cpu", threadMXBean.getCurrentThreadUserTime() - threadUserTime)
+          );
```

Comment:
It provides User time and general CPU time. I can subtract the two to get "non user time" but didn't feel that was terribly useful since it couldn't be broken down anymore.

There are MANY types of cpu times reported by tools like `htop` which cannot be bubbled up through the jvm easily ( or consistently across host OS)